nupic/embodied/utils/model_parts.py

```diff
@@ -0,0 +1,243 @@
+        device,
+        batchnorm=False,
+    ):
+        super(small_convnet, self).__init__()
+        self.device = device
+        # Get the input shape
+        self.H = ob_space.shape[0]
+        self.W = ob_space.shape[1]
+        self.C = ob_space.shape[2]
+        # Attributes of the convolutional layers
+        feat_list = [[self.C, 32, 8, (4, 4)], [32, 64, 8, (2, 2)], [64, 64, 3, (1, 1)]]
+        self.conv = torch.nn.Sequential().to(device)
+        oH = self.H
+        oW = self.W
+        for idx, f in enumerate(feat_list):
+            # Add convolutional layer
+            self.conv.add_module(
+                "conv_%i" % idx,
+                torch.nn.Conv2d(f[0], f[1], kernel_size=f[2], stride=f[3]),
+            )
+            # Apply nonlinear activation function
+            self.conv.add_module("nl_%i" % idx, nonlinear())
+            if batchnorm:
+                # Normalize batch
+                self.conv.add_module("bn_%i" % idx, torch.nn.BatchNorm2d(f[1]))
+            # Calculations to get flat output dimensionality of last conv layer
+            oH = (oH - f[2]) / f[3][0] + 1
+            oW = (oW - f[2]) / f[3][1] + 1
+
+        assert oH == int(oH)  # whether oH is a .0 float ?
+        assert oW == int(oW)
+
+        self.flatten_dim = int(oH * oW * feat_list[-1][1])
+        # Add fc layer at end for feature output
+        self.fc = torch.nn.Linear(self.flatten_dim, feat_dim).to(device)
+
+        self.last_nonlinear = last_nonlinear
+        self.layernormalize = layernormalize
+        # initialize weight with xavier uniform distribution
+        self.init_weight()
+
+    def init_weight(self):
+        """Initialize wieght of network."""
+        for m in self.conv:
+            if isinstance(m, torch.nn.Conv2d):
+                torch.nn.init.xavier_uniform_(m.weight.data)
+                torch.nn.init.constant_(m.bias.data, 0.0)
+        torch.nn.init.xavier_uniform_(self.fc.weight.data)
+        torch.nn.init.constant_(self.fc.bias.data, 0.0)
+
+    def forward(self, x):
+        """Forward pass of x through the network.
+
+        Parameters
+        ----------
+        x : array
+            input (observations).
+
+        Returns
+        -------
+        array
+            Output features of the network.
+
+        """
+        # run x through the convolutional layers
+        x = self.conv(x)
+        # Get flattened version of conv output
+        # TODO: check that contiguous.view actually does what its supposed to
+        x = x.contiguous().view(
+            -1, self.flatten_dim
+        )  # dims is calculated manually, 84*84 -> 20*20 -> 9*9 ->7*7
+        # run through fully connected layer
+        x = self.fc(x)
+        if self.last_nonlinear is not None:
+            # apply activation function the the last layer
+            x = self.last_nonlinear(x)
+        if self.layernormalize:
+            # normalize the last layer
+            x = self.layernorm(x)
+        return x
```

Comment:
If we change all operations to be layer, we can make the classifier be a sequential as well, so `self.classifier = nn.Sequential(nn.Flatten, fc, layernorm)` and then just call self.classifier(x)